### ISSUE TITLE
Wrong Imports in Generated `.d.ts` on Windows

### ISSUE THREAD

**Issue by yfwz100** [wessberg/rollup-plugin-ts#21]
As the picture shown, the generated file `index.esm.d.ts` contains OS-specific path imports:

![image](https://user-images.githubusercontent.com/983211/58770370-1d773c00-85e1-11e9-945e-768562168f65.png)

More importantly, `Suggestion` interface is redefined...

**Comment by yfwz100** [comment]
It works fine under OS X. Seems like a problem related to path.

**Comment by wessberg** [comment]
Hi there, thanks for reporting this issue.
I think this issue and https://github.com/wessberg/rollup-plugin-ts/issues/19 are related.

Looking at the image, the import paths look wrong, so I'm certain that the issue is related to how relative paths are computed by the plugin.

I'll try to spin up a Windows VM and test it

**Comment by yfwz100** [comment]
Is there any code determining whether a symbol is bundled or imported? As the pictures shows, the `Suggestion` interface is declared twice.

**Comment by wessberg** [comment]
Yes there is, but grouped by emitted chunk, and since the import paths are broken, the logic for determining whether or not symbols are defined is broken on Windows

**Comment by yfwz100** [comment]
Any help I can try? Not familiar with code but the plugin is awesome. It's better than the official one.

**Comment by Septh** [comment]
I work on a Windows 10 box and I'd like to offer my help, too, should you need some.

I just switched to rollup-plugin-ts to build [my own plugin](https://github.com/Septh/rollup-plugin-node-externals). For now, I'm using a separate build step to generate the declaration file and I'd rather have your plugin do all the dirty job :) in one step.

I've tried taking a look at your source code but it's dense and hard to follow when not used to. Just give some directions as to where to look in first.

**Comment by wessberg** [comment]
I'm investigating the issue, and the issue is related to a couple of things:

- When comparing two paths or filenames, `rollup-plugin-ts` always assume that they follow POSIX. For example, there are strict equality between generated filenames for Rollup chunks, which uses the path convention of the OS, and Typescript sourcefile filenames, which are normalized to POSIX. This means that some comparisons fail that wouldn't on OS X where everything is POSIX.

- The `path` module is used heavily to normalize file paths. However, they aren't normalized to POSIX when generating new `ImportDeclarations` and `ExportDeclarations` which means that stuff like `import {foo} from "\\a\\b\\c"` can happen.

I'm working on fixing these things and I'm testing both on Mac OS and Windows. I'll keep you posted.

**Comment by wessberg** [comment]
The issue has been fixed and released in v1.1.55. Thanks for reporting it!
